### PR TITLE
Avoid unnecessary transform resets for pixel snapping

### PR DESCRIPTION
**Pull request by flar**
During some new development work that might make transform resets more expensive we realized that the resets were mostly coming from the calls to snap the transform to a pixel translate value and many of those were NOPs since the transform was already on a pixel translate value. This PR will avoid those trivially unnecessary reset operations.

### CODE REVIEW

**Comment by flar** [comment]
Discovered while working on https://github.com/flutter/engine/pull/50935

File: flow/raster_cache_util.cc [hunk 1]
```diff
@@ -4,4 +4,67 @@
...
+    return true;
+  }
+
+  return false;
+}
+
+bool RasterCacheUtil::ComputeIntegralTransCTM(const SkM44& in, SkM44* out) {
+  // Avoid integral snapping if the matrix has complex transformation to avoid
+  // the artifact observed in https://github.com/flutter/flutter/issues/41654.
+  if (in.rc(0, 1) != 0 || in.rc(0, 2) != 0) {
```
Comment (jonahwilliams) [inline flow/raster_cache_util.cc:36]: rc = row column? Its quite the API...

Comment (jonahwilliams) [reply flow/raster_cache_util.cc:36]: (I realize this is from Skia)

File: flow/raster_cache_util.cc [hunk 2]
```diff
@@ -4,4 +4,67 @@
...
+    *out = in;
+    (*out)[SkMatrix::kMTransX] = out_tx;
+    (*out)[SkMatrix::kMTransY] = out_ty;
+    return true;
+  }
+
+  return false;
+}
+
+bool RasterCacheUtil::ComputeIntegralTransCTM(const SkM44& in, SkM44* out) {
```
Comment (jonahwilliams) [inline flow/raster_cache_util.cc:33]: Do we have any tests that are exercising the SkM44 code path?

**Review by jonahwilliams — APPROVED** [review]
Overall LGTM but I'd like it if there was an SkM44 test that we needed to update too...

**Comment by flar** [comment]
There were added to fix problems we had with not maintaining full 4x4 transforms during the layer recursion so they are tested someplace indirectly. but, I could add some LSS/DlMCT tests to test them more directly.

**Review by knopp — APPROVED** [review]

**Pull request merged**